### PR TITLE
feat(ion-header): add basic UI for ionic theme

### PR DESCRIPTION
**Pull request by BenOsodrac**
Issue number: internal

---------

## What is the new behavior?

- Added new common header partial, to be used by ionic theme and ios/md
- Renamed header.scss to header.native.scss. This inherits the common created above and its used only on ios and md partial
- Created new partial for ionic theme.
- Added ionic theme to basic e2e tests.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

[Sample Header Screen](https://ionic-framework-rcfex2fep-ionic1.vercel.app/src/components/header/test/basic?ionic:theme=ionic)

### CODE REVIEW

**Review by OS-giulianasilva — APPROVED** [review]

File: core/src/components/header/header.ionic.scss
Comment (thetaPC) [inline core/src/components/header/header.ionic.scss:10]: Are we expecting the box shadow to change? If not, then let's follow the structure that is established for `md`:

```suggestion
  .header-ionic {
     box-shadow: #{globals.$ionic-elevation-200};
  }
```

File: core/src/components/header/test/basic/header.e2e.ts
Comment (thetaPC) [inline core/src/components/header/test/basic/header.e2e.ts:4]: The box shadow isn't being shown. Let's update this test to capture that style:

```
await page.setContent(
  `
  <style>
    .container {
      background-color: #eaeaea;
      padding: 10px;
    }
  </style>
v
  <div class="container">
    <ion-header>
      <ion-toolbar>
        <ion-title>Header - Default</ion-title>
      </ion-toolbar>
    </ion-header>
  </div>
`,
  config
);

const container = page.locator('.container');
await expect(container).toHaveScreenshot(screenshot(`header-diff`));
```

**Review by thetaPC — CHANGES_REQUESTED** [review]

File: core/src/components/header/header.ionic.scss
Comment (thetaPC) [inline core/src/components/header/header.ionic.scss:12]: This needs to be moved outside of `ion-header` for it to work. By placing it inside, it targets `ion-header > .header-ionic`.

**Review by thetaPC — CHANGES_REQUESTED** [review]

File: core/src/components/header/header.ionic.scss
Comment (BenOsodrac) [reply core/src/components/header/header.ionic.scss]: good catch, thanks 👍

**Review by thetaPC — APPROVED** [review]
LGTM, also thank you for separating the files into native and common!!!

The snapshots for the ionic theme had to be regenerated. There are moments that the pixel difference is so small that Playwright doesn't recognize the change. In this case, it was the box shadow. I regenerated the snapshots for you so you're not waiting on an approval while I'm out.

**Comment by BenOsodrac** [comment]
> LGTM, also thank you for separating the files into native and common!!!
> 
> The snapshots for the ionic theme had to be regenerated. There are moments that the pixel difference is so small that Playwright doesn't recognize the change. In this case, it was the box shadow. I regenerated the snapshots for you so you're not waiting on an approval while I'm out.

Ah thank you! 🖤 

No prob, separating the files does result in cleaner code no doubt :)

**Pull request merged**
